titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java

```diff
@@ -258,59 +313,106 @@ public void setVMLeaseHandler(Action1<List<? extends VirtualMachineLease>> lease
         this.leaseHandler = leaseHandler;
     }
 
+    @Override
+    public void setRescindLeaseHandler(Action1<List<LeaseRescindedEvent>> rescindLeaseHandler) {
+        this.rescindLeaseHandler = rescindLeaseHandler;
+    }
+
     @Override
     public Observable<LeaseRescindedEvent> getLeaseRescindedObservable() {
-        return vmLeaseRescindedObserver.asObservable();
+        return PublishSubject.create();
     }
 
     @Override
     public Observable<ContainerEvent> getTaskStatusObservable() {
         return vmTaskStatusObserver.asObservable();
     }
 
-    private void pollForNodes() {
+    private void getNodeUpdates() {
+        try {
+            long startTimeMs = clock.wallTime();
+            V1NodeList list = null;
+            try {
+                list = normalApi.listNode(null, null, null, null, null, null, null, null, null);
+                nodesClientMetrics.incrementOnSuccess(GET, NODES, STATUS_200);
+            } catch (Exception e) {
+                logger.error("Failed to list nodes with error:", e);
+                nodesClientMetrics.incrementOnError(GET, NODES, e);
+            } finally {
+                nodesClientMetrics.registerLatency(GET, startTimeMs);
```

Comment:
Do we want to associate the latency with whether the request was successful or not?

Reply:
Yes, these metrics were just for getting preliminary latency numbers despite the http code. Going to convert the client to using `WebClient` with the general metrics code.